### PR TITLE
Issue/2550 clean dfjq regularly (Connect #2550)

### PR DESCRIPTION
**Pull request by stellanl**
#### Before the PR (what is the issue or what needed to be done)
Entries stayed in the DFJQ forever
#### The solution
Check them once a week, delete those fulfilled or >2 years old.
## Checklist
* [ ] Connect the issue
* [ ] Test plan
* [ ] Copyright header
* [ ] Code formatting
* [ ] Documentation

### CODE REVIEW

File: GAE/src/org/waterforpeople/mapping/app/web/CronCommanderServlet.java [hunk 1]
```diff
@@ -58,7 +65,52 @@ public void doGet(HttpServletRequest req, HttpServletResponse resp)
...
+        }
+    }
+
+    /**
+     * scans for and deletes DeviceFileJobQueue entries that are either more than two years old
+     * or that refer to files that have been successfully uploaded.
+     */
+    private void purgeDeviceFileJobQueueRecords() {
+        Calendar deadline = Calendar.getInstance();
+        deadline.add(Calendar.YEAR, -2); //two years ago
```
Comment (valllllll2000) [inline GAE/src/org/waterforpeople/mapping/app/web/CronCommanderServlet.java]: Why not extract the constant to a meaningful name and remove the comment?

File: GAE/src/org/waterforpeople/mapping/app/web/CronCommanderServlet.java [hunk 2]
```diff
@@ -58,7 +65,52 @@ public void doGet(HttpServletRequest req, HttpServletResponse resp)
...
+                                item.getKey().getId());
+                        retirees++;
+                    }
+                } catch (Exception e) {
+                    // catchall
+                    log.warning("Error while connecing to " + item.getFileName() +"\n\n" + e.getMessage());
+                }
+            }
         }
+        log.info("Attempted to retire "+retirees+" of " + dfjqList.size());
```
Comment (valllllll2000) [inline GAE/src/org/waterforpeople/mapping/app/web/CronCommanderServlet.java]: it seems to be there are spaces missing, shouldn't it be ...retire " + retirees + "

File: GAE/src/org/waterforpeople/mapping/app/web/CronCommanderServlet.java [hunk 3]
```diff
@@ -58,7 +65,52 @@ public void doGet(HttpServletRequest req, HttpServletResponse resp)
...
+                    if (conn.getResponseCode() == 200) {
+                        // best case - fulfilled
+                        log.info("Deleting fulfilled DFJQ entry: " + item.getKey().getId());
+                        SurveyTaskUtil.spawnDeleteTask(SurveyTaskRequest.DELETE_DFJQ_ACTION,
+                                item.getKey().getId());
+                        retirees++;
+                    }
+                } catch (Exception e) {
+                    // catchall
+                    log.warning("Error while connecing to " + item.getFileName() +"\n\n" + e.getMessage());
```
Comment (valllllll2000) [inline GAE/src/org/waterforpeople/mapping/app/web/CronCommanderServlet.java]: Why 2 newlines?

File: GAE/src/org/waterforpeople/mapping/app/web/CronCommanderServlet.java [hunk 4]
```diff
@@ -58,7 +65,52 @@ public void doGet(HttpServletRequest req, HttpServletResponse resp)
...
+                            " : " + conn.getResponseCode() + " " + conn.getResponseMessage());
+                    if (conn.getResponseCode() == 200) {
+                        // best case - fulfilled
+                        log.info("Deleting fulfilled DFJQ entry: " + item.getKey().getId());
+                        SurveyTaskUtil.spawnDeleteTask(SurveyTaskRequest.DELETE_DFJQ_ACTION,
+                                item.getKey().getId());
+                        retirees++;
+                    }
+                } catch (Exception e) {
+                    // catchall
```
Comment (valllllll2000) [inline GAE/src/org/waterforpeople/mapping/app/web/CronCommanderServlet.java]: is this comment necessary?

File: GAE/src/org/waterforpeople/mapping/app/web/SurveyTaskServlet.java [hunk 5]
```diff
@@ -77,37 +78,34 @@ protected RestResponse handleRequest(RestRequest req) throws Exception {
...
-                SurveyTaskRequest.DELETE_QUESTION_GROUP_ACTION)) {
+        } else if (stReq.getAction().equals(SurveyTaskRequest.DELETE_QUESTION_GROUP_ACTION)) {
             QuestionGroupDao qgDao = new QuestionGroupDao();
             QuestionGroup qg = qgDao.getByKey(id);
             if(qg != null) {
                 qgDao.delete(qg);
             }
-        } else if (stReq.getAction().equals(
-                SurveyTaskRequest.DELETE_QUESTION_ACTION)) {
+        } else if (stReq.getAction().equals(SurveyTaskRequest.DELETE_QUESTION_ACTION)) {
```
Comment (valllllll2000) [inline GAE/src/org/waterforpeople/mapping/app/web/SurveyTaskServlet.java]: 2 things:
* wouldn't it make sense to extract stReq.getAction() to a local variable at the beginning? action = stReq.getAction();
* what if stReq.getAction() == null? I would use SurveyTaskRequest.DELETE_QUESTION_ACTION.equals(action)

File: GAE/src/org/waterforpeople/mapping/app/web/SurveyTaskServlet.java [hunk 6]
```diff
@@ -69,77 +70,86 @@ protected RestRequest convertRequest() throws Exception {
...
-                                        stReq.getId().toString()).param(
-                                        SurveyTaskRequest.ID_LIST_PARAM,
-                                        buffer.toString()).param(
-                                        SurveyTaskRequest.CURSOR_PARAM,
-                                        SurveyInstanceDAO.getCursor(nextSet)));
                     }
                 }
-            }
+                break;
+            default:log.warning("Unknown action.");
```
Comment (valllllll2000) [inline GAE/src/org/waterforpeople/mapping/app/web/SurveyTaskServlet.java]: oups, needs a newline after default:

File: GAE/src/org/waterforpeople/mapping/app/web/SurveyTaskServlet.java [hunk 7]
```diff
@@ -69,77 +70,88 @@ protected RestRequest convertRequest() throws Exception {
...
+                SurveyDAO surveyDao = new SurveyDAO();
+                Survey s = surveyDao.getByKey(id);
+                if(s != null) {
+                    surveyDao.delete(s);
+                }
+                break;
+            case SurveyTaskRequest.DELETE_QUESTION_GROUP_ACTION:
+                QuestionGroupDao qgDao = new QuestionGroupDao();
+                QuestionGroup qg = qgDao.getByKey(id);
+                if(qg != null) {
```
[the remark this answers is not in the document]
Comment (stellanl) [reply GAE/src/org/waterforpeople/mapping/app/web/SurveyTaskServlet.java]: Those were there already, but ok...

**Pull request merged**